devtools/conda-envs/examples_env.yml

```
@@ -14,7 +14,7 @@ dependencies:
     - packaging
     # Examples
     - openff-toolkit-examples>=0.15.2
-    - openff-interchange>=0.3.18
+    - openff-interchange ~=0.3.25
```

```suggestion
    - openff-interchange >=0.3.25
```

I think >= is actually the desired semantics - we want this environment not to need updating if an incompatible version of Interchange is released. There isn't any assumption that code will remain compatible with Interchange, because any code that uses Interchange comes either from the selected version of Interchange or a compatible version of a different package.

I can see that `~=` is more appropriate for a downloadable environment for end-users, but the environment file is checked daily and isn't really designed for users to hold on to long term before they solve it. Users should re-download the new example and get the updated environment rather than use an old download.

What do you think?

---

IIUC, `~=` will pull down any 0.3.x that's released in the future but never pull down the 0.4.0 that I'm hoping exists at some point in the future.

In this case the difference may not be so big if 0.4.x doesn't break any examples, which is my hope, and if I'm going to be manually updating this anyway, which is more or less my expectation.

---

Yeah ok totally fair. Sorry, I think I was worried that I'd be updating pins every time a minor release came out, but if it's your job to fix your examples then it's totally fair that you get to choose how to manage your releases!